kpatch-build/create-diff-object.c

```diff
@@ -1063,27 +1114,14 @@ static void kpatch_correlate_static_local_variables(struct kpatch_elf *base,
 				ERROR("sections %s and %s aren't correlated for symbol %s",
 				      sym->sec->name, patched_sym->sec->name, sym->name);
 
-			log_debug("renaming and correlating static local %s to %s\n",
-				  patched_sym->name, sym->name);
-
-			patched_sym->name = strdup(sym->name);
-			sym->twin = patched_sym;
-			patched_sym->twin = sym;
-
-			/* set initial status, might change */
-			sym->status = patched_sym->status = SAME;
+			kpatch_correlate_static_local(sym, patched_sym);
 
 			if (bundled) {
-				sym->sec->twin = patched_sym->sec;
-				patched_sym->sec->twin = sym->sec;
+				kpatch_correlate_section(sym->sec, patched_sym->sec);
+				kpatch_correlate_symbol(sym->sec->secsym, patched_sym->sec->secsym);
 
-				sym->sec->secsym->twin = patched_sym->sec->secsym;
-				patched_sym->sec->secsym->twin = sym->sec->secsym;
-
-				if (sym->sec->rela && patched_sym->sec->rela) {
-					sym->sec->rela->twin = patched_sym->sec->rela;
-					patched_sym->sec->rela->twin = sym->sec->rela;
-				}
+				if (sym->sec->rela && patched_sym->sec->rela)
+					kpatch_correlate_section(sym->sec->rela, patched_sym->sec->rela);
```

Comment:
This is all quite similar to what kpatch_process_section_correlation() does... could it all be replaced with
```
if (bundled) {
        kpatch_correlate_section(sym->sec, patched_sym->sec);
        kpatch_process_section_correlation(sym->sec);
}
```
?

And for that matter I wonder if it would be better to have kpatch_correlate_section() automatically do all the work in kpatch_process_section_correlation()?  Then kpatch_correlate_section() could call a lower-level __kpatch_correlate_section() version which doesn't do that extra work.  So kpatch_correlate_section() would be a higher-level interface which just does the right thing, and then this could just be
```
if (bundled)
        kpatch_correlate_section(sym->sec, patched_sym->sec);
```
And similarly kpatch_correlate_sections() wouldn't need to call kpatch_process_section_correlation().

Reply:
Yes, having kpatch_correlate_section() do all the needed work seems like a good suggestion.
Thanks.